connector_magento_multi_source_inventory/models/magento_source/stock_exporter.py

```diff
@@ -0,0 +1,84 @@
+                    "parent_id": picking.external_id,
+                    "weight": 0,
+                    "qty": 1,
+                    "description": picking.name,
+                    "track_number": picking.carrier_tracking_ref,
+                    "title": picking.carrier_id.magento_tracking_title,
+                    "carrier_code": picking.carrier_id.magento_carrier_code,
+                }
+            }]
+        return (picking.carrier_id.magento_carrier_code,
+                picking.carrier_id.magento_tracking_title or '',
+                picking.carrier_tracking_ref)
+
+    def _validate(self, binding):
+        if binding.state != 'done':  # should not happen
+            raise ValueError("Wrong value for picking state, "
+                             "it must be 'done', found: %s" % binding.state)
+        if not binding.carrier_id.magento_carrier_code:
+            raise FailedJobError("Wrong value for the Magento carrier code "
+                                 "defined in the picking.")
+
+    def _check_allowed_carrier(self, binding, external_id):
+        """ Magento2 API does not allow to fetch the list of allowed carriers.
+        """
+        if self.collection.version == '2.0':
+            return
+        allowed_carriers = self.backend_adapter.get_carriers(external_id)
+        carrier = binding.carrier_id
+        if carrier.magento_carrier_code not in allowed_carriers:
+            raise FailedJobError("The carrier %(name)s does not accept "
+                                 "tracking numbers on Magento.\n\n"
+                                 "Tracking codes accepted by Magento:\n"
+                                 "%(allowed)s.\n\n"
+                                 "Actual tracking code:\n%(code)s\n\n"
+                                 "Resolution:\n"
+                                 "* Add support of %(code)s in Magento\n"
+                                 "* Or deactivate the export of tracking "
+                                 "numbers in the setup of the carrier "
+                                 "%(name)s." %
+                                 {'name': carrier.name,
+                                  'allowed': allowed_carriers,
+                                  'code': carrier.magento_carrier_code})
+
+    def run(self, binding):
+        """ Export the stock level to Magento """
+
+        binder = self.binder_for()
+        external_id = binder.to_external(binding)
+        if not external_id:
+            picking_exporter = self.component(usage='record.exporter')
+            picking_exporter.run(binding)
+            external_id = binder.to_external(binding)
+        if not external_id:
+            return FailedJobError("The delivery order %s has no Magento ID, "
+                                  "can't export the tracking number." %
+                                  binding.name)
+
+        self._validate(binding)
+        self._check_allowed_carrier(binding, external_id)
```

Comment:
Maybe you could remove all this carrier related stuff first, to make it more clear (I can imagine this also makes the tests fail).